massa-execution/src/vm.rs

```diff
@@ -174,6 +177,63 @@ impl VM {
         context.ledger_step.caused_changes.clone()
     }
 
+    /// Run code in read-only mode
+    pub(crate) fn run_read_only(
+        &self,
+        slot: Slot,
+        max_gas: u64,
+        simulated_gas_price: Amount,
+        bytecode: Vec<u8>,
+        address: Option<Address>,
+        result_sender: oneshot::Sender<ExecuteReadOnlyResponse>,
+    ) {
+        let saved_changes = {
+            let mut context = self.execution_context.lock().unwrap();
```

Comment:
use self.clear_and_update_context(); (see the run_step_internal function) to initialize the context, otherwise it will simply be inherited from the last call. This way the rest of the context will also be reset to a default state here, to ensure that if we add fields to the context, there is no persistence between calls in case we forget to port them here

Reply:
Ok thanks, done.